### ISSUE TITLE
. mjs configuration file error

### ISSUE THREAD

**Issue by ajiho** [antonk52/lilconfig#56]
pkg.json

```
  "scripts": {
    "test": "node scripts/test.mjs"
  },
```

scripts\test.mjs

```
import { lilconfig } from "lilconfig";

const res = await lilconfig("myapp").search(); // Promise<LilconfigResult>

console.log(res);
```

Create myapp.config.mjs in the project root directory

```
npm run test
```
```
> node scripts/test.mjs

node:internal/modules/esm/load:227
    throw new ERR_UNSUPPORTED_ESM_URL_SCHEME(parsed, schemes);
          ^

Error [ERR_UNSUPPORTED_ESM_URL_SCHEME]: Only URLs with a scheme in: file, data, and node are supported by the default ESM loader. On Windows, absolute paths must be valid file:// URLs. Received protocol 'c:'
    at throwIfUnsupportedURLScheme (node:internal/modules/esm/load:227:11)
    at defaultLoad (node:internal/modules/esm/load:109:3)
    at ModuleLoader.load (node:internal/modules/esm/loader:666:12)
    at ModuleLoader.loadAndTranslate (node:internal/modules/esm/loader:479:43)
    at #createModuleJob (node:internal/modules/esm/loader:503:36)
    at #getJobFromResolveResult (node:internal/modules/esm/loader:274:34)
    at ModuleLoader.getModuleJobForImport (node:internal/modules/esm/loader:242:41)
    at async onImport.tracePromise.__proto__ (node:internal/modules/esm/loader:542:25) {
  code: 'ERR_UNSUPPORTED_ESM_URL_SCHEME'
}

Node.js v22.11.0
```

**Comment by antonk52** [comment]
Hi and thanks for the report. Unfortunately I am not able to repro this.

Seeing the backward slashes for the script in your message and the mention of `c:` in the error message I presume that you are using Windows OS. I do not have a windows machine to test this. My assumption is that the dynamic import gets a file path like `c:\path\to\myapp.config.mjs` and takes `c:` parts as a path scheme. Which is not the case for Windows OS. In your node modules do you mind adding `console.log(id)` here and re-running your script to see the full path that it outputs?

If this is correct, this should be fairly easy to fix. We can add `const url = require('url')` to the top of lilconfig/src/index.js and replace the [dynamic import line](https://github.com/antonk52/lilconfig/blob/2cb3e756e1e1d890caee88d3f44a898c7903b2a2/src/index.js#L57) with this
```diff
-		const mod = await import(/* webpackIgnore: true */ id);
+		const fileUrl = url.pathToFileURL(id).href;
+		const mod = await import(/* webpackIgnore: true */ fileUrl);
```

You can try this out by manually applying this change in your project's node_modules/lilconfig/src/index.js and rerunning your script.

Please let me know if this fixes the issue for you? In this case I am happy to release a new version that fixes this.

**Comment by antonk52** [comment]
The fixed is published to npm in v3.1.3

Thanks for the report
